### ISSUE TITLE
Issue with Node Duplication in Apache Age When Creating Relations

### ISSUE THREAD

**Issue by gunjan-it-engg** [apache/age#1340]
**The bug**
Issue with Node Duplication in Apache Age When Creating Relations

Hi Apache Age Team,

I hope this message finds you well. I've come across an issue when creating relations between nodes in Apache Age. In a scenario where I establish relations like A->B and A->C, I've observed that Apache Age is creating duplicate nodes for A. This behavior differs from what we typically experience in Neo4j.

In Neo4j, this relation creation would not result in node duplication. However, in Apache Age, it seems to generate multiple instances of node A.

This deviation from expected behavior could potentially impact data consistency and performance in some use cases. I wanted to bring this to your attention and seek guidance or clarification on whether this behavior is intentional or if there's a workaround to prevent node duplication.

I appreciate your support and look forward to your insights on this matter.

Best regards,
Gunjan Shrimali

**Comment by jrgemignani** [comment]
@gunjan-it-engg Why was this issue closed? Was it in error?

**Comment by gunjan-it-engg** [comment]
> @gunjan-it-engg Why was this issue closed? Was it in error?

"Previously, when I executed a Cypher query to create nodes and set their properties while also establishing relationships between them, it resulted in the creation of duplicate nodes. The query looked like this:

```
SELECT * FROM ag_catalog.cypher('building', $$
CREATE (:Building {name: 'building1', id: "B1"})-[:Has_FLOOR]->(:Floor {name: 'Ground', id: "G"})
$$) AS (a ag_catalog.agtype);
```
Upon running this query, it would create two nodes with identical properties and labels. To resolve this issue, I modified my approach. Now, I first create the nodes with their properties, and then I establish relationships between them. This adjustment has successfully eliminated the problem of duplicate nodes."
